test/test_smbo/test_smbo.py

```diff
@@ -50,7 +50,8 @@ class TestSMBO(unittest.TestCase):
     def setUp(self):
         self.scenario = Scenario({'cs': test_helpers.get_branin_config_space(),
                                   'run_obj': 'quality',
-                                  'output_dir': 'data-test_smbo'})
+                                  'output_dir': 'data-test_smbo',
+                                  "runcount-limit": 5})
         self.output_dirs = []
```

Comment:
Was there a need to include a "runcount-limit"?

Reply:
If the `TestCallback` would not return `False` at some point, or if stopping the optimization loop would not work, the test would run forever. This is not the case, but I thought it'd be better to have this safety net. I'm not sure if this has an impact on the other tests. Let me know if you would like me to remove this, or constrain it to `test_incorporate_run_results_callback_stop_loop`.

Reply:
This makes sense to me but hopefully someone more familiar with all the tests can decide.

Reply:
Looks good to me